dut/drivers.wake

```diff
@@ -0,0 +1,59 @@
+
+tuple SkeletalDriverPlan =
+  global DuhFile:             Path
+  global CompanyName:         String
+  global DeviceName:          String
+  global LocalMetalDirectory: String
+  global DUTPlan:             RocketChipDUTPlan
+
+global def makeSkeletalDriverPlan = SkeletalDriverPlan _ _ _ _ _
+
+def getDevicePlan deviceName =
+  def plans = subscribe skeletalDriverPlans
+  def samePlan aPlan =
+    if aPlan.getSkeletalDriverPlanDeviceName ==~ deviceName
+    then Some aPlan
+    else None
+  mapPartial samePlan plans
+  | head
```

Comment:
This feels like an awkward construction in Wake, since it obfuscates what I believe is your intent. I think you're just trying to find the first item in a list that matches a predicate, so you should just use the [`find`](https://github.com/sifive/wake/blob/a91f5e84f780fb620adb0c6ad14cf80114c2dc46/share/wake/lib/core/list.wake#L139) function.

Reply:
fixed